code/modules/error_handler/error_handler.dm

```diff
@@ -0,0 +1,97 @@
+/var/list/error_last_seen = list()
+// error_cooldown items will either be positive (cooldown time) or negative (silenced error)
+//  If negative, starts at -1, and goes down by 1 each time that error gets skipped
+/var/list/error_cooldown = list()
+/var/total_runtimes = 0
+/var/total_runtimes_skipped = 0
+// The ifdef needs to be down here, since the error viewer references total_runtimes
+#ifdef DEBUG
+/world/Error(var/exception/e)
+	if (!istype(e)) // Something threw an unusual exception
+		world.log << "\[[time_stamp()]] Uncaught exception: [e]"
+		return ..()
+
+	if (!global.error_last_seen) // A runtime is occurring too early in start-up initialization
+		return ..()
+
+	global.total_runtimes++
+
+	var/erroruid = "[e.file][e.line]"
+	var/last_seen = global.error_last_seen[erroruid]
+	var/cooldown = global.error_cooldown[erroruid] || 0
```

Comment:
Is || 0 even necessary? I thought [] threw an error on bad access, rather than return null.
I don't even know what the behaviour is if you only have bad values in a ||